lib/mixlib/shellout/windows.rb

```diff
@@ -313,6 +320,38 @@ def self.find_executable(path)
         def self.executable?(path)
           File.executable?(path) && !File.directory?(path)
         end
+
+        def self.kill_process_tree(pid, wmi, logger)
+          wmi.query("select * from Win32_Process where ParentProcessID=#{pid}").each do |instance|
```

Comment:
It looks like this block is designed to not be run when we get to the edges of the tree, i.e. a process has no children, because the WMI query will return an empty set. Maybe there's value in making that more obvious with something like:

```
children = wmi.query("select * from Win32_Process where ParentProcessID=#{pid}")
if children.any?
  logger.debug(["found a bunch of children, going to kill them"])
  children.each do |instance|
    stuff
  end
else
  logger.debug(["no children"])
end
```

Not sure. At least a comment here saying "this method kills all children if they exist or just quits otherwise." or whatever.